File: app/src/main/java/ru/vtb/ppcd/example/route/impl/InnerRouteTo.java
package ru.vtb.ppcd.example.route.impl;

import ru.vtb.ppcd.example.config.ApplicationConfig;
import ru.vtb.ppcd.example.config.RouteId;
import lombok.RequiredArgsConstructor;
import org.apache.camel.builder.RouteBuilder;
import org.springframework.stereotype.Component;


@Component
@RequiredArgsConstructor
public class InnerRouteTo extends RouteBuilder {
    private final ApplicationConfig applicationConfig;

    @Override
    public void configure() {
        var routePath = applicationConfig.getRoutePathWithIdById(getRouteId());
        from(routePath.in())
                .id(getRouteId().name())
                .to(routePath.out());
    }

    private RouteId getRouteId() {
        return RouteId.DIRECT_OUT_TOPIC_ID;
    }
}

File: ms-ppcd-example-app/src/main/kotlin/ru/vtb/ppcd/example/route/impl/InnerRouteTo.kt
package org.example.ru.vtb.ppcd.example.route.impl

import org.apache.camel.builder.RouteBuilder
import org.example.ru.vtb.ppcd.example.config.RouteConfig
import org.example.ru.vtb.ppcd.example.model.RouteId
import org.springframework.stereotype.Component

@Component
class InnerRouteTo(
    private val routeConfig: RouteConfig,
) : RouteBuilder() {
    private val routeId: RouteId = RouteId.DIRECT_OUT_TOPIC_ID

    override fun configure(): Unit =
        routeConfig.getRoutePathWithIdById(routeId)
            .let { routePath ->
                from(routePath.`in`)
                    .id(routePath.routeId.name)
                    .to(routePath.out)
            }
}